sympy/matrices/expressions/determinant.py

```diff
@@ -50,7 +57,11 @@ def det(matexpr):
     1
     """
 
-    return Determinant(matexpr).doit()
+    if isinstance(matexpr, MatrixExpr) and kwargs:
+        raise ValueError(
+            "Keyword arguments not supported for matrix expressions.")
```

Comment:
I think this should be in the `_eval_determinant` of `MatrixExpr`, as it's specific to `MatrixExpr` and one can call `doit` directly. Also, it only should check for `method` keyword argument.

Reply:
Please see my previous comment "https://github.com/sympy/sympy/pull/13857#issuecomment-359150385".

Reply:
I think signatures of all `_eval_determinant` should be changed to accept `*kwargs`. Also, ideally `kwargs` check should be in each `_eval_determinant`, but it seems that this cannot be centralized unless it's done here.

Reply:
Move this check to `doit()`, as one can call `doit()` directly.

Reply:
If we move this check, we have to add `matexpr` argument in `doit()` function definition, which I think might be redundant. I think it's better to keep as it is.
What do you say @normalhuman ?

Reply:
@ArighnaIITG You don't need to add `matexpr` argument in `doit()`. `Determinant` already has `matexpr`.